### PR TITLE
Add breakingChanges for httpChecksum trait

### PR DESCRIPTION
**Pull request by gosar**
#### Background
The `aws.protocols#httpChecksum` trait was missing any rules for detecting backwards incompatible changes to the applied trait. This PR adds `breakingChanges` to the trait definition.

#### Testing
Added tests to assert the events for diff.

#### Links

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy [hunk 1]
```diff
@@ -123,7 +123,42 @@ structure ec2Query {}
 string ec2QueryName
 
 /// Indicates that an operation supports checksum validation.
-@trait(selector: "operation")
+@trait(
+    selector: "operation",
+    breakingChanges: [
+        {
+            change: "remove",
+            severity: "ERROR",
```
Comment (hpmellema) [inline smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:131]: Error is the default severity for breaking changes, so we can probably remove that here.

File: smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy [hunk 2]
```diff
@@ -123,7 +123,42 @@ structure ec2Query {}
...
+                `requestAlgorithmMember` was removed so newly generated clients will no longer be able to pick the \
+                request checksum algorithms The service MUST continue to support old clients by supporting \
+                `requestAlgorithmMember`."""
+        },
+        {
+            change: "remove",
+            path: "/requestValidationModeMember",
+            severity: "ERROR",
+            message: """
+                `requestValidationModeMember` was removed so newly generated clients will no longer validate response \
```
Comment (hpmellema) [inline smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:147]:
```suggestion
                `requestValidationModeMember` was removed, so newly generated clients will no longer validate response \
```

Comment (hpmellema) [reply smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:147]: Same above.

File: smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy (hunk 1 above)
Comment (yefrig) [inline smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:131]: Should we include a message when the entire trait is removed?

**Review by yefrig — APPROVED** [review]

File: smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy (hunk 1 above)
Comment (hpmellema) [reply smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:131]: I noticed that the request compression rules use `DANGER` [see](https://github.com/smithy-lang/smithy/blob/e3348620935216be2943f92c5288ec0a6762353e/smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy#L1253). These seem like they would have the same severity.

Comment (gosar) [reply smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:131]: I debated it and felt removal of compression being less severe than removal of checksums was reasonable. But don't feel strongly about it, so changing to DANGER.

Comment (gosar) [reply smithy-aws-traits/src/main/resources/META-INF/smithy/aws.protocols.smithy:131]: oh I wanted to add message later after adding test, but forgot. thanks for catching.

**Review by hpmellema — APPROVED** [review]

**Pull request merged**